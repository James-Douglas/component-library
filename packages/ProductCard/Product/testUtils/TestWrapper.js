import React from 'react';
import ProductProvider from '../../Context/ProductProvider';

const ProductTestWrapper = ({
  // eslint-disable-next-line react/prop-types
  children, product, onDetails = () => {}, onSelect = () => {},
}) => (
  <ProductProvider product={product} onDetails={onDetails} onSelect={onSelect}>
    { children }
  </ProductProvider>
);

export default ProductTestWrapper;
