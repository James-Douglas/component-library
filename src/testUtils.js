// test-utils.js
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './providers/ManorProvider';

const customRender = (ui, options) => render(ui, { wrapper: ManorProvider, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
