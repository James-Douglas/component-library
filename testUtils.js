import React from 'react';
import {
  render, fireEvent, wait, act,
} from '@testing-library/react';
import ManorProvider from './packages/Provider/ManorProvider';

const ManorTestWrapper = ({ children }) => (
  <ManorProvider>
    {children}
  </ManorProvider>
);

const customRender = (ui, options) => render(ui, { wrapper: ManorTestWrapper, ...options });

// re-export everything
export * from '@testing-library/react';

// override render method
export {
  customRender as render,
  // export other used functions to appease eslint
  fireEvent,
  wait,
  act,
};
