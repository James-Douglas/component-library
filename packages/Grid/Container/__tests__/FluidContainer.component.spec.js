import React from 'react';
import { render } from '../../../../testUtils';
import 'jest-styled-components';
import FluidContainer from '../FluidContainer.component';

describe('Container', () => {
  it('renders correctly without props', () => {
    const { container } = render(<FluidContainer />);
    expect(container).toMatchSnapshot();
  });

  it('renders children', () => {
    const { container } = render(
      <FluidContainer>
        <div className="child-content">test</div>
      </FluidContainer>,
    );

    const childContent = container.querySelector('.child-content');
    expect(container).toMatchSnapshot();
    expect(childContent).toBeInTheDocument();
  });

  it('has a container class by default and NO fixed class', () => {
    const { container } = render(<FluidContainer />);

    const containerComponent = container.children[1];

    expect(container).toMatchSnapshot();
    expect(containerComponent).toBeDefined();
    expect(containerComponent).toHaveStyleRule('padding-left', '1.6rem');
    expect(containerComponent).not.toHaveStyleRule('max-width', '72rem');
  });

  it('allows additional classes to be passed through', () => {
    const { container } = render(<FluidContainer className="test-class" />);

    expect(container).toMatchSnapshot();
    const test = container.getElementsByClassName('test-class');
    expect(test).toBeDefined();
  });
});
