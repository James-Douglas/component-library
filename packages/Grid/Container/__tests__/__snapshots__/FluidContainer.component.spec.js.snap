// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container allows additional classes to be passed through 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  margin: auto;
  padding: 0;
}

<div>
  <div
    class="c0 test-class"
  />
</div>
`;

exports[`Container allows padding to be passed through 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  margin: auto;
  padding: 0 0.4rem;
}

<div>
  <div
    class="c0 test-class"
  />
</div>
`;

exports[`Container has a container class by default and NO fixed class 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  margin: auto;
  padding: 0;
}

<div>
  <div
    class="c0"
  />
</div>
`;

exports[`Container renders children 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  margin: auto;
  padding: 0;
}

<div>
  <div
    class="c0"
  >
    <div
      class="child-content"
    >
      test
    </div>
  </div>
</div>
`;

exports[`Container renders correctly without props 1`] = `
.c0 {
  box-sizing: border-box;
  width: 100%;
  margin: auto;
  padding: 0;
}

<div>
  <div
    class="c0"
  />
</div>
`;
